### DISCUSSION TITLE
Get rules, with nested, as array

### DISCUSSION THREAD

**Discussion by bianchi** [spatie/laravel-data#495] [Q&A]
I want trying to generate some automatic documentation for my endpoints with [Scribe](https://scribe.knuckles.wtf/), but using laravel-data in my requests. For most endpoints it's working but in the ones I have nested rules I need to get all rules at once as an array.

So what I want to achieve is this:

```
[
  'groups' => ['array', 'required', 'min:1'],
  'groups.*.name' => ['required', 'string', 'max:45']
]
```

In my parent class I have:

```
<?php

declare(strict_types=1);

namespace App\Datas;

use Spatie\LaravelData\Attributes\DataCollectionOf;
use Spatie\LaravelData\Data;
use Spatie\LaravelData\DataCollection;


class CreateMultipleGroupsRequest extends Data
{
    public function __construct(
        #[DataCollectionOf(StoreGroupRequest::class)]
        public DataCollection $groups,
    ) {
    }

    public static function rules(): array
    {
        return [
            'groups' => ['array', 'required', 'min:1'],
        ];
    }
}
```

while in the child:

```
<?php

declare(strict_types=1);

namespace App\Datas;

use Spatie\LaravelData\Data;

class StoreGroupRequest extends Data
{
    public function __construct(
        public string $name
    ) {
    }

    public static function rules(): array
    {
        return [
            'name' => ['required', 'string', 'max:45']
        ];
    }
}
```

If I call CreateMultipleGroupsRequest::rules(), as expected, I only get:
```
[
  'groups' => ['array', 'required', 'min:1'],
];
```

If I call CreateMultipleGroupsRequest::getValidationRules([]) I get:

```
array:2 [ // app/Docs/Strategies/GetFromBodyLaravelData.php:49
  "groups" => array:3 [
    0 => "array"
    1 => "required"
    2 => "min:1"
  ]
  "groups.*" => Illuminate\Validation\NestedRules^ {#3576
    #callback: Closure(mixed $value, mixed $attribute)^ {#3575
      class: "Spatie\LaravelData\Resolvers\DataValidationRulesResolver"
      this: Spatie\LaravelData\Resolvers\DataValidationRulesResolver {#3533 …}
      use: {
        $fullPayload: []
        $dataProperty: Spatie\LaravelData\Support\DataProperty {#3583 …}
      }
    }
  }
]
```

I don't have the payload here to pass to `getValidationRules`.

Any ideas on how to get the desired array?

**Comment by rubenvanassche** [comment]
I don't think there's an easy solution, data's validation is rewritten in v3 to write rules based upon the provided data. You could use the DataClass and DataProperty classes for each data class and try to generate a payload based upon the information there with empty arrays like the example above.

Good luck!

**Comment by dizews** [comment]
@bianchi hello. How are you using scribe for laravel-data objects? Do you configured special strategies? could you show me an example of you integration?

**Reply by bianchi** [reply]
I wrote custom strategies as in scribe docs [Scribe](https://scribe.knuckles.wtf/laravel/reference/plugin-api)

**Reply by dizews** [reply]
could you be so kind to share with me?

**Reply by bianchi** [reply]
There are several files. Call me on twitter to not pollute this issue https://twitter.com/jhonatanbianchi

**Reply by dizews** [reply]
I follow you on twitter. nick name is the same as here.

**Reply by igorsgm** [reply]
@bianchi @dizews Would you guys mind to also share it with me? 🙏
